airbytehq/airbyte
# otel metric client implementation

## What
[#12908 ]Implement an OTEL metric client

## How
- Introduce OTEL dependency
- Install OTEL SDK and OTEL Meter object in initialization; Use count/gauge/histogram for count/gauge/distribution implementation.
- Manual test
  -  Start an OTEL Collector locally, attached screenshot is what I collected. Note the name and description matched with the original metric, and service.name is the actual metric app name.
  ![Screen Shot 2022-06-03 at 6 10 22 PM](https://user-images.githubusercontent.com/5505742/171970727-61ea20e8-73b1-4c0e-9c84-97f5d97cbe67.png)
  - Manually inject gauge/distribution functions into worker code as test, and verify those functions can be called correctly.
